### PR TITLE
Add tests for WITH RECURSIVE use cases

### PR DESCRIPTION
**Pull request by kavinderd**
This commit introduces the gp_recursive_cte test file with a variety of
tests concentrated on exercising referencing a recursive CTE in a
subquery

### CODE REVIEW

File: src/test/regress/greenplum_schedule [hunk 1]
```diff
@@ -114,7 +114,7 @@ test: bfv_cte bfv_joins bfv_subquery bfv_planner bfv_legacy bfv_temp
 
 test: qp_olap_mdqa qp_misc
```
Comment (hsyuan) [inline src/test/regress/greenplum_schedule]: can we put the test here?

Comment (kavinderd) [reply src/test/regress/greenplum_schedule]: Yeah sure, I wasn't sure where to add it

File: src/test/regress/sql/gp_recursive_cte.sql [hunk 2]
```diff
@@ -0,0 +1,83 @@
+-- Tests exercising different behaviour of the WITH RECURSIVE implementation in GPDB
+-- GPDB's distributed nature requires thorough testing of many use cases in order to ensure correctness
+
+-- Setup
```
Comment (hsyuan) [inline src/test/regress/sql/gp_recursive_cte.sql]: create a schema with name of gp_recursive_cte and add it to search_path. and drop table is not needed at the end of the test.

Comment (ashwinstar) [reply src/test/regress/sql/gp_recursive_cte.sql]: Using proper table (object) names, little more test specific helps in general, plus also eliminates need to schema or dropping tables. Also, helps to run tests in parallel as needed.

**Comment by kavinderd** [comment]
@ashwinstar thanks for noticing that. I forgot to add the test file. I just added the expected file and added more meaningful test names

**Comment by dhanashreek89** [comment]
LGTM. Could you please also add a test case for Expression Sublink for completeness ?
For instance:
```
with recursive r(i) as (
   select 1
   union all
   select i + 1 from r
)
select * from recursive_table_1 where recursive_table_1.id >= (select i from r limit 1);
```

**Comment by kavinderd** [comment]
@vraghavan78  Since master is currently not in a state for new commits I will push this branch to this repo so additional tests can be opened as separate PR's agains that branch

**Pull request closed without merging**